### PR TITLE
Mesh 2176/fix-storage-naming

### PR DESCRIPTION
**Pull request by HenriqueNogara**
### modified external API

- Deprecate the following asset storage: `AssetMetadataNextLocalKey`,  `AssetMetadataNextGlobalKey`;
- Deprecate the following NFT storage: `NextCollectionId`, `NextNFTId`;

### new external API

- Add the following asset storage: `CurrentAssetMetadataLocalKey`, `CurrentAssetMetadataGlobalKey`;
- Add the following NFT storage: `CurrentNFTId`, `CurrentCollectionId`;

### data migration
- Initializes new storage for the asset pallet;
- Initializes new storage for the nft pallet;

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -1362,7 +1380,8 @@ impl<T: Config> Module<T> {
         );
 
         // Next global key.
-        let key = AssetMetadataNextGlobalKey::try_mutate(try_next_pre::<T, _>)?;
+        let key = Self::update_current_asset_metadata_global_key()?;
+        AssetMetadataNextGlobalKey::try_mutate(try_next_pre::<T, _>)?;
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: Just set the old storage value to `key`.  The mutate operation has an extra read that we don't need.

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -2311,7 +2330,8 @@ impl<T: Config> Module<T> {
         );
 
         // Next local key for asset.
-        let key = AssetMetadataNextLocalKey::try_mutate(ticker, try_next_pre::<T, _>)?;
+        let key = Self::update_current_asset_metadata_local_key(&ticker)?;
+        AssetMetadataNextLocalKey::try_mutate(ticker, try_next_pre::<T, _>)?;
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: Same here.  Just set the old storage value to the same value `key`.

File: pallets/asset/src/lib.rs [hunk 3]
```diff
@@ -2524,3 +2581,33 @@ impl<T: Config> AssetFnTrait<T::AccountId, T::RuntimeOrigin> for Module<T> {
         Self::add_mandatory_mediators(origin, ticker, mediators.try_into().unwrap_or_default())
     }
 }
+
+pub mod migration {
+    use sp_runtime::runtime_logger::RuntimeLogger;
+
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::IterableStorageMap;
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::StorageMap;
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::StorageValue;
```
Comment (Neopallium) [inline pallets/asset/src/lib.rs]: Please don't import these from a "hidden" crate, they should be available from a public crate.

File: pallets/nft/src/lib.rs [hunk 4]
```diff
@@ -295,7 +303,8 @@ impl<T: Config> Module<T> {
         }
 
         // Creates the nft collection
-        let collection_id = NextCollectionId::try_mutate(try_next_pre::<T, _>)?;
+        let collection_id = Self::update_current_collection_id()?;
+        NextCollectionId::try_mutate(try_next_pre::<T, _>)?;
```
Comment (Neopallium) [inline pallets/nft/src/lib.rs]: Use set instead of mutate for the old storage.

File: pallets/nft/src/lib.rs [hunk 5]
```diff
@@ -360,7 +369,8 @@ impl<T: Config> Module<T> {
         let new_balance = NumberOfNFTs::get(&ticker, &caller_portfolio.did)
             .checked_add(1)
             .ok_or(Error::<T>::BalanceOverflow)?;
-        let nft_id = NextNFTId::try_mutate(&collection_id, try_next_pre::<T, _>)?;
+        let nft_id = Self::update_current_nft_id(&collection_id)?;
+        NextNFTId::try_mutate(&collection_id, try_next_pre::<T, _>)?;
```
Comment (Neopallium) [inline pallets/nft/src/lib.rs]: Same here.

File: pallets/nft/src/lib.rs [hunk 6]
```diff
@@ -626,22 +668,30 @@ impl<T: Config> NFTTrait<T::RuntimeOrigin> for Module<T> {
...
-    use crate::sp_api_hidden_includes_decl_storage::hidden_include::IterableStorageDoubleMap;
-    use crate::{Config, NFTOwner};
-    use frame_support::storage::StorageDoubleMap;
-    use pallet_portfolio::PortfolioNFT;
     use sp_runtime::runtime_logger::RuntimeLogger;
 
-    pub fn migrate_to_v2<T: Config>() {
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::IterableStorageMap;
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::StorageMap;
+    use crate::sp_api_hidden_includes_decl_storage::hidden_include::StorageValue;
```
Comment (Neopallium) [inline pallets/nft/src/lib.rs]: Same here.  Don't use hidden imports.

**Review by Neopallium — CHANGES_REQUESTED** [review]

**Review by Neopallium — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**